tests/integration/test_logcollector/test_options/test_restrict_ignore_regex.py

```diff
@@ -0,0 +1,195 @@
+        - set_wazuh_configuration_with_local_internal_options:
+            type: fixture
+            brief: Set the wazuh configuration according to the configuration data and local_internal_options.
+        - restart_wazuh_function:
+            type: fixture
+            brief: Restart wazuh.
+
+    assertions:
+        - Check that logcollector is analyzing the log file.
+        - Check that logs are ignored when they do not match with configured regex
+
+    input_description:
+        - The `configuration_ignore_regex_default.yaml` file provides the module configuration for this test.
+        - The `cases_ignore_regex_default` file provides the test cases.
+
+    expected_output:
+        - r".*wazuh-logcollector.*Analizing file: '{file}'.*"
+        - r".*wazuh-logcollector.*DEBUG: Reading syslog '{message}'.*"
+        - r".*wazuh-logcollector.*DEBUG: Ignoring the log line '{message}' due to {tag} config: '{regex}'"
+    '''
+    log = metadata['log_sample']
+    command = f"echo {log}>> {test_file}"
+
+    if sys.platform == 'win32':
+        file = re.escape(test_file)
+    else:
+        file = test_file
+
+    # Check log file is being analized
+    evm.check_analyzing_file(file=file, prefix=prefix)
+
+    #  Insert log
+    run_local_command_returning_output(command)
+    # Check the log is read from the monitored file
+    evm.check_syslog_messages(message=log, prefix=prefix)
+
+    # Check responses
+    # If it matches with ignore, it should ignore the log due to ignore config
+    if 'ignore' in metadata['matches']:
+        evm.check_ignore_restrict_messages(message=log, regex=metadata['ignore_regex'], tag='ignore',
+                                           prefix=prefix)
+        if 'restrict' in metadata['matches']:
+            log_found = False
+            with pytest.raises(TimeoutError):
+                log_found = evm.check_ignore_restrict_messages(message=log, regex=metadata['restrict_regex'],
+                                                               tag='restrict', prefix=prefix)
+            assert log_found is False, lc.ERR_MSG_UNEXPECTED_IGNORE_EVENT
+    # If matches with restrict, it should not be ignored due to restrict config
+    elif metadata['matches'] == 'restrict':
+        log_found = False
+        with pytest.raises(TimeoutError):
+            log_found = evm.check_ignore_restrict_messages(message=log, regex=metadata['restrict_regex'],
+                                                           tag='restrict', prefix=prefix)
+        assert log_found is False, lc.ERR_MSG_UNEXPECTED_IGNORE_EVENT
+        log_found = False
+        with pytest.raises(TimeoutError):
+            log_found = evm.check_ignore_restrict_messages(message=log, regex=metadata['ignore_regex'],
+                                                           tag='ignore', prefix=prefix)
+        assert log_found is False, lc.ERR_MSG_UNEXPECTED_IGNORE_EVENT
```

Comment:
Should the log error show 2 times? 
Couldn't we check all in the same section? 
We will check it in a call.

Reply:
We are checking that the log does not appear either with restrict or ignore (Neither should appear.